expression/builtin_arithmetic_vec.go

```diff
@@ -107,11 +107,110 @@ func (b *builtinArithmeticDivideDecimalSig) vecEvalDecimal(input *chunk.Chunk, r
 }
 
 func (b *builtinArithmeticModIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinArithmeticModIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	x := result.Int64s()
+	y := buf.Int64s()
+	xIsunsiged := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	yIsunsiged := mysql.HasUnsignedFlag(b.args[1].GetType().Flag)
+	switch {
+	case xIsunsiged && yIsunsiged:
+		for i := 0; i < n; i++ {
+			if buf.IsNull(i) {
+				result.SetNull(i, true)
```

Comment:
1. Use MergeNulls outside of switch-case instead of SetNull.
2. The check of `result.IsNull(i)` can be moved before the check of `y[i] == 0`

Reply:
```go
func (s *builtinArithmeticModIntSig) evalInt(row chunk.Row) (val int64, isNull bool, err error) {
	b, isNull, err := s.args[1].EvalInt(s.ctx, row)
	if isNull || err != nil {
		return 0, isNull, err
	}

	if b == 0 {
		return 0, true, handleDivisionByZeroError(s.ctx)
	}

	a, isNull, err := s.args[0].EvalInt(s.ctx, row)
	if isNull || err != nil {
		return 0, isNull, err
	}
```
In this unvectoried function , it's logic determines  that we can't use mergernull outside of switch-case. If we do this way ,it way decreases number of warnings .